granulate_utils/config_feeder/client/bigdata/dataproc.py

```diff
@@ -0,0 +1,41 @@
+from typing import Any, Dict, Optional
+
+import requests
+from requests.exceptions import ConnectionError, JSONDecodeError
+
+from granulate_utils.config_feeder.core.models.cluster import BigDataPlatform, CloudProvider
+from granulate_utils.config_feeder.core.models.node import NodeInfo
+
+
+def get_dataproc_node_info() -> Optional[NodeInfo]:
+    """
+    https://cloud.google.com/compute/docs/metadata/querying-metadata
+    https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/metadata
+
+    tested on Dataproc 2.0-debian10
+    """
+    try:
+        metadata = _get_metadata()
+        attributes = metadata["attributes"]
+        properties = {
+            "region": attributes["dataproc-region"],
+            "cluster_name": attributes["dataproc-cluster-name"],
+        }
+        return NodeInfo(
+            external_id=str(metadata["id"]),
+            external_cluster_id=attributes["dataproc-cluster-uuid"],
+            is_master=attributes["dataproc-role"] == "Master",
+            provider=CloudProvider.GCP,
+            bigdata_platform=BigDataPlatform.DATAPROC,
+            properties=properties,
+        )
+    except (ConnectionError, JSONDecodeError, KeyError):
+        return None
```

Comment:
Should we log specific errors? For example, if the connection was refused we can just return None but if we expect the request to return a JSON data response and we encounter a parsing error let's log it, same for KeyError